collector/exporter/otelarrowexporter/internal/arrow/stream.go

```diff
@@ -460,11 +482,28 @@ func (s *Stream) SendAndWait(ctx context.Context, records interface{}) error {
 			return err
 		}
 	}
+	// Note that the uncompressed size as measured by the receiver
+	// will be different than uncompressed size as measured by the
+	// exporter, because of the optimization phase performed in the
+	// conversion to Arrow.
+	var uncompSize int
+	switch data := records.(type) {
+	case ptrace.Traces:
+		var sizer ptrace.ProtoMarshaler
+		uncompSize = sizer.TracesSize(data)
```

Comment:
What is the cost of this function call? Is it called by default?

Reply:
Good catch. The call is not free, so I've made them conditional on `MetricsLevel > LevelNormal` so this only costs when the user asks for extra metrics.